db/db.go

```diff
@@ -0,0 +1,238 @@
+package db
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"go_final_project/models"
+	"log"
+	"os"
+	"time"
+
+	_ "modernc.org/sqlite"
+)
+
+var DB *sql.DB
+
+func ConnectDB() {
+	// *2
+	// Определяем путь к файлу базы данных через переменную окружения
+	dbFile := os.Getenv("TODO_DBFILE")
+	if dbFile == "" {
+		dbFile = "scheduler.db"
+	}
+
+	var install bool
+	if _, err := os.Stat(dbFile); err != nil {
+		if os.IsNotExist(err) {
+			install = true
+			fmt.Println("database does not exist")
+		} else {
+			panic(err)
+		}
+	} else {
+		fmt.Println("database already exists")
+	}
+
+	// Открытие базы данных
+	db, err := sql.Open("sqlite", dbFile)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	DB = db
+
+	if install {
+		// Создание базы данных и индекса
+		CreateScheduler := `CREATE TABLE scheduler (
+            id INTEGER PRIMARY KEY AUTOINCREMENT,
+            date CHAR(8) NOT NULL DEFAULT "",
+            title TEXT NOT NULL DEFAULT "",
+            comment TEXT,
+            repeat VARCHAR(128) NOT NULL DEFAULT ""
+        );`
+		_, err := DB.Exec(CreateScheduler)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		// Создание индекса по полю date
+		createIndex := `CREATE INDEX date_scheduler ON scheduler(date);`
+		_, err = DB.Exec(createIndex)
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println("database created successfully")
+	}
+}
+
+// Добавляем задачу в БД
+func AddTask(date, title, comment, repeat string) (int64, error) {
+	fmt.Fprintln(os.Stdout, "request modify data", date, title, comment, repeat)
+	res, err := DB.Exec("INSERT INTO scheduler (date, title, comment, repeat) VALUES (:date, :title, :comment, :repeat)",
+		sql.Named("date", date),
+		sql.Named("title", title),
+		sql.Named("comment", comment),
+		sql.Named("repeat", repeat))
+
+	if err != nil {
+		return 0, err
+	}
+
+	id, err := res.LastInsertId()
+	if err != nil {
+		return 0, err
```

Comment:
Лучше все ошибки оборачивать, чтобы быстрее понимать, где они произошли
```suggestion
		return 0, fmt.Errorf("error in getting id: %w", err)
```